### PR TITLE
test(e2e): validate the reported diff against ArgoCD's own rendering

### PR DESCRIPTION
**Pull request by shini4i**
The lab proved expansion parity but never checked the comparison result — `smoke` asserts the diff agrees with a hand-written expectation, not with ArgoCD. This closes that: ArgoCD renders the addon chart at each branch's commit, those renders are diffed, and argo-compare's reported diff must match in both directions. Every change ArgoCD sees must appear in the diff, and the diff must invent none.

Two findings shaped the design, both worth knowing:

`argocd app manifests --revision` renders chart content at a revision using the Application spec **held in the cluster**, so it cannot model a change to the manifest itself — asked for two revisions differing only in inline values, ArgoCD returns identical output. The addon Application therefore keeps an identical manifest on both branches with no inline values, leaving chart content as the only thing that can move its render.

Its chart also needed a `.argo-compare.yml`: a chart-only change with no manifest change reaches argo-compare no other way. That is correct behaviour and the reason anchors exist, and it makes this phase the strictest test of the anchor flow.

ArgoCD re-serialises manifests through its object model, so helm's `message: "hello"` comes back as `message: hello`; `normalize_diff` strips that quoting before the sets are compared. `section_diff` is now bounded by the next `===>` heading — a mutation revealed it had been collecting later Applications' diffs.

Verified by a clean-room `task e2e` from a deleted cluster, all three phases green. Mutation-checked: pointed at the wrong Application's section, both directions fail.

Stacked on #168.

### CODE REVIEW

File: test/e2e/scripts/render-parity.sh [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+main_sha="$(git -C "${work}/gitops" rev-parse "origin/${TARGET_BRANCH}")"
+feature_sha="$(git -C "${work}/gitops" rev-parse "origin/${FEATURE_BRANCH}")"
+[[ "$main_sha" != "$feature_sha" ]] || die "both branches point at the same commit"
+
+argocd_login || die "could not log in to the lab's ArgoCD"
+
+argocd_manifests addon "$main_sha" >"${work}/argo-main.yaml" ||
+  die "argocd could not render addon at ${main_sha}"
+argocd_manifests addon "$feature_sha" >"${work}/argo-feature.yaml" ||
+  die "argocd could not render addon at ${feature_sha}"
```
[the remark this answers is not in the document]
Comment (shini4i) [reply test/e2e/scripts/render-parity.sh:29]: Confirmed and fixed in 1c25112. `task up` waited for the applicationset-controller and repo-server but not `argocd-server`, so the readiness you describe was genuinely unguarded — my clean-room runs only passed because seeding happened to give the server time.

Both calls are now retried, booting ArgoCD waits for the server rollout alongside the controllers, and a render attempt writes to a file which must be non-empty before it counts, so a partial failure cannot reach the caller as though it were the full output.

Verified on a cold boot: `task down` then `task e2e` from no cluster, all three phases green.

File: test/e2e/scripts/lib.sh [hunk 2]
```diff
@@ -119,6 +119,58 @@ build_compare() {
...
+
+# section_diff <output> <marker>: the changed lines of one Application's section
+# of an argo-compare run, which reports every Application in one stream. Bounded
+# by the next `===>` heading, so a section that is not the last one does not
+# collect its successors' diffs.
+section_diff() {
+  local output="$1" marker="$2"
+  awk -v m="$marker" '
+    /^===>/ { inside = index($0, m) ? 1 : 0; next }
+    inside && /^[+-][^+-]/ { print }
```
[the remark this answers is not in the document]
Comment (shini4i) [reply test/e2e/scripts/lib.sh:188]: Fair, and fixed in 1c25112. `lib.bats` covers the helpers offline — whitespace stripping and sorting, the scalar unquoting that reconciles helm's output with ArgoCD's re-serialisation, deduplication, section boundaries, an absent section, and diff_body on identical files. `task lint` runs it, so these no longer need a cluster to surface a bug.

Writing them was worthwhile beyond the coverage: piping into a filter via `run bash -c` does not carry a function sourced by `setup()`, so one case was passing on a command-not-found message rather than on the helper's output. Each case goes through a helper in the test shell now.

Mutation-checked: removing the unquoting fails four cases, and unbounding `section_diff` fails the boundary case.
